### PR TITLE
fix: added Loader and Error Handling for TrustPay GooglePay

### PR DESCRIPTION
**Pull request by ArushKapoorJuspay**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

1. Added Background Loader once TrustPay GooglePay button is clicked.
2. Added polling and timeout for TrustPay GooglePay

## How did you test it?

Tested transactions via TrustPay GooglePay

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/orca-loader/Elements.res [hunk 1]
```diff
@@ -300,40 +300,74 @@ let make = (
...
+                          ~switchToCustomPod,
+                          ~isForceSync=true,
+                        )
+                      )
+                    let executeGooglePayment = trustpay.executeGooglePayment(
+                      payment,
+                      googlePayRequest,
+                    )
+                    let timeOut =
+                      Utils.delay(600000)->then(_ => resolve(Dict.make()->JSON.Encode.object))
```
Comment (vsrivatsa-juspay) [inline src/orca-loader/Elements.res:338]: Is this supposed to be 1m or 10m?

File: src/orca-loader/Elements.res [hunk 2]
```diff
@@ -300,40 +300,74 @@ let make = (
...
+
+                    Promise.race([polling, executeGooglePayment, timeOut])
+                    ->then(res => {
+                      logger.setLogInfo(
+                        ~value="TrustPay GooglePay Success Response",
```
Comment (vsrivatsa-juspay) [inline src/orca-loader/Elements.res:343]: If timeout is resolved first, the log shouldn't be TrustPay GooglePay Success Response

Comment (vsrivatsa-juspay) [reply src/orca-loader/Elements.res:343]: Resolve timeOut promise with a failure string.
Decode res and check if it matches the failure string, and branch out the handling

**Review by vsrivatsa-juspay — CHANGES_REQUESTED** [review]
Please resolve comments

**Review by prafulkoppalkar — APPROVED** [review]

**Pull request merged**